src/views/tutorial-view/utils/get-product-data.ts

```diff
@@ -0,0 +1,23 @@
+import fs from 'fs'
+import path from 'path'
+import moize from 'moize'
+import { LearnProductData, LearnProductSlug } from 'types/products'
+
+function getProductData(product: LearnProductSlug): LearnProductData {
+	try {
+		const productData = JSON.parse(
+			fs.readFileSync(
+				path.join(process.cwd(), `src/data/${product}.json`),
+				'utf-8'
+			)
+		)
+		return productData
+	} catch (e) {
+		console.error(
+			`[Error]: unable to fetch product data for ${product} — ${e.message}`
+		)
+		throw e
+	}
+}
```

Comment:
So I don't think this will work during ISR, as the file system isn't kept around. An option could be to use a dynamic import, as that would include all of the json files as JS modules and they would be included in the deployed artifact.

Reply:
Ah okay I originally had it as a a dynamic import but I've read about that being less performant, perhaps because of what you mentioned with all the json files being included. I'll look into it, forgot about the fs / ISR issues 💀

Reply:
Breadcrumb per slack. TL:DR vercel may support using FS in ISR now - https://nextjs.org/docs/advanced-features/output-file-tracing